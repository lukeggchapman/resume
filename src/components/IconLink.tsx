import React from 'react'
import { Text, Link as PDFLink } from '@react-pdf/renderer'
import FAIcon, { FAIconProps } from 'components/FAIcon'

export type IconLinkProps = FAIconProps & {
  src: string
  children: React.ReactNode
}

const IconLink: React.FC<IconLinkProps> = ({ src, children, ...iconProps }) => (
  <Text>
    <FAIcon {...iconProps} />
    {` `}
    <PDFLink src={src}>{children}</PDFLink>
  </Text>
)

export default IconLink
