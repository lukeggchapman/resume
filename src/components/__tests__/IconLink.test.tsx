import React from 'react'
import { shallow } from 'enzyme'

import IconLink, { IconLinkProps } from '../IconLink'

const baseProps: IconLinkProps = {
  src: 'srcTest',
  children: 'childrenTest',
  far: true,
  icon: 'envelope',
}

describe('IconLink', () => {
  const render = () => shallow(<IconLink {...baseProps} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
