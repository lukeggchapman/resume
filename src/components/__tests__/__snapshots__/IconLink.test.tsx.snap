// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink renders correctly 1`] = `
<Text>
  <FAIcon
    far={true}
    icon="envelope"
  />
   
  <Link
    src="srcTest"
  >
    childrenTest
  </Link>
</Text>
`;
